(v5) WithContext types are troublesome to mock

I've been experimenting with the Context aware v5.0.0 preview and I've ran into some issues when trying to write unit tests.

In version 4.X of the Go package interface types such as `Result`, `Session` and `Transaction` have exported methods only. Which can easily be mocked using tools such as [vektra/mockery](https://github.com/vektra/mockery). However, when using version 5.0.0-preview, the new context aware types `ResultWithContext`, `SessionWithContext` and `TransactionWithContext` contain new unexported methods such as `legacy()` which in some cases return unexported types such as `*result`. 

Whilst `mockery` can still create mocks for the interfaces, I am unable to use the mock `ResultWithContext` as the `legacy()` method has an unexported return value  `*result`.

I get the following error when building using the mock of `ResultWithContext`:
```
mocks/ResultWithContext.go:398:46: result not exported by package neo4j
```

Would it be possible to remove the unexported methods on those interfaces, or more importantly, remove the unexported return value `*result`.


If you would like to generate the mock yourself, run the following:
```bash
$ go install github.com/vektra/mockery/v2@latest
$ mockery --with-expecter --exported --dir $(go list -m -f '{{ .Dir }}' github.com/neo4j/neo4j-go-driver/v5)/neo4j --name ResultWithContext --output mocks  
```

--- a/neo4j/result.go
+++ b/neo4j/result.go
@@ -57,7 +57,7 @@ type result struct {
 }
 
 // deprecated: use newResultWithContext instead
-func newResult(conn db.Connection, str db.StreamHandle, cypher string, params map[string]interface{}) *result {
+func newResult(conn db.Connection, str db.StreamHandle, cypher string, params map[string]interface{}) Result {
 	return newResultWithContext(conn, str, cypher, params).legacy()
 }
 

--- a/neo4j/result_with_context.go
+++ b/neo4j/result_with_context.go
@@ -53,7 +53,8 @@ type ResultWithContext interface {
 	Consume(ctx context.Context) (ResultSummary, error)
 	// IsOpen determines whether this result cursor is available
 	IsOpen() bool
-	legacy() *result
+	buffer(ctx context.Context)
+	legacy() Result
 }
 
 const consumedResultError = "result cursor is not available anymore"
@@ -72,7 +73,7 @@ type resultWithContext struct {
 }
 
 func newResultWithContext(conn idb.Connection, str idb.StreamHandle,
-	cypher string, params map[string]interface{}) *resultWithContext {
+	cypher string, params map[string]interface{}) ResultWithContext {
 	return &resultWithContext{
 		conn:         conn,
 		streamHandle: str,
@@ -199,7 +200,7 @@ func (r *resultWithContext) IsOpen() bool {
 	return r.isOpen()
 }
 
-func (r *resultWithContext) legacy() *result {
+func (r *resultWithContext) legacy() Result {
 	return &result{delegate: r}
 }
 

--- a/neo4j/transaction_with_context.go
+++ b/neo4j/transaction_with_context.go
@@ -160,7 +160,7 @@ func (tx *managedTransaction) legacy() Transaction {
 // Does not implement the ExplicitTransaction nor the ManagedTransaction interface.
 type autocommitTransaction struct {
 	conn     db.Connection
-	res      *resultWithContext
+	res      ResultWithContext
 	closed   bool
 	onClosed func()
 }

